Wikipedia/Code/HelpViewController.swift

```diff
@@ -38,13 +56,43 @@ class HelpViewController: SinglePageWebViewController {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
     }
+    
+    private func setupToolbarItems(isExportingUserData: Bool) {
+        
+        let exportItem = isExportingUserData ? spinnerToolbarItem : exportUserDataToolbarItem
+        if isExportingUserData {
+            activityIndicator.startAnimating()
+        } else {
+            activityIndicator.stopAnimating()
+        }
+        
+        let leadingSpace: CGFloat = isExportingUserData ? 30 : 8
+        
+        self.toolbar.items = [UIBarButtonItem.wmf_barButtonItem(ofFixedWidth: leadingSpace), exportItem, UIBarButtonItem.flexibleSpaceToolbar(), sendEmailToolbarItem, UIBarButtonItem.wmf_barButtonItem(ofFixedWidth: 8)]
+    }
 
     private func setupToolbar() {
         enableToolbar()
-        toolbar.items = [UIBarButtonItem.flexibleSpaceToolbar(), sendEmailToolbarItem, UIBarButtonItem.wmf_barButtonItem(ofFixedWidth: 8)]
+        setupToolbarItems(isExportingUserData: false)
         setToolbarHidden(false, animated: false)
     }
     
+    enum UserDataExportError: Error {
+        case unableToDetectSystemFreeSpace
+        case unableToDetectLogSize
+    }
+    
+    @objc func exportUserData() {
+        
+        let confirmationTitle = WMFLocalizedString("export-user-data-confirmation-title", value: "Share app library?", comment: "Title of confirmation modal after user taps \"Export User Data\" button.")
+        let confirmationMessage = WMFLocalizedString("export-user-data-confirmation-message", value: "Sharing your app library includes data on your Reading lists and history, preferences and Explore feed content. This data file should only be shared with a trusted recipient to use for technical diagnostic purposes.", comment: "Message of confirmation modal after user taps \"Export User Data\" button.")
```

Comment:
I saw it in the Phab ticket, but I'd recommend a minor copy update to this string:

`Sharing your app library includes data about your Reading lists and history, preferences, and Explore feed content. This data file should only be shared with a trusted recipient to use for technical diagnostic purposes.`